pkg/deploy/controller/deployment/controller.go

```diff
@@ -24,40 +39,48 @@ import (
 //
 //   1. If the deployment finished normally, the deployer pod is deleted.
 //   2. If the deployment failed, the deployer pod is not deleted.
-//
-// Use the DeploymentControllerFactory to create this controller.
 type DeploymentController struct {
-	// serviceAccount to create deployment pods with
+	// rn is used for updating replication controllers.
+	rn kclient.ReplicationControllersNamespacer
+	// pn is used for creating, updating, and deleting deployer pods.
+	pn kclient.PodsNamespacer
+
+	// queue contains replication controllers that need to be synced.
+	queue workqueue.RateLimitingInterface
+
+	// rcStore is a store of replication controllers.
+	rcStore cache.StoreToReplicationControllerLister
+	// podStore is a store of pods.
+	podStore cache.StoreToPodLister
+	// rcStoreSynced makes sure the rc store is synced before reconcling any deployment.
+	rcStoreSynced func() bool
+	// podStoreSynced makes sure the pod store is synced before reconcling any deployment.
+	podStoreSynced func() bool
+
+	// deployerImage specifies which Docker image can support the default strategies.
+	deployerImage string
+	// serviceAccount to create deployment pods with.
 	serviceAccount string
-	// deploymentClient provides access to deployments.
-	deploymentClient deploymentClient
-	// podClient provides access to pods.
-	podClient podClient
-	// makeContainer knows how to make a container appropriate to execute a deployment strategy.
-	makeContainer func(strategy *deployapi.DeploymentStrategy) *kapi.Container
-	// decodeConfig knows how to decode the deploymentConfig from a deployment's annotations.
-	decodeConfig func(deployment *kapi.ReplicationController) (*deployapi.DeploymentConfig, error)
+	// environment is a set of environment variables which should be injected into all
+	// deployer pod containers.
+	environment []kapi.EnvVar
+	// codec is used for deserializing deploymentconfigs from replication controller
+	// annotations.
+	codec runtime.Codec
 	// recorder is used to record events.
 	recorder record.EventRecorder
 }
 
-// fatalError is an error which can't be retried.
-type fatalError string
-
-func (e fatalError) Error() string { return "fatal error handling deployment: " + string(e) }
-
-// actionableError is an error on which users can act
-type actionableError string
-
-func (e actionableError) Error() string { return string(e) }
-
 // Handle processes deployment and either creates a deployer pod or responds
 // to a terminal deployment status.
 func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) error {
```

Comment:
> godoc whether this RC is an original or a copy

I think I prefer accepting the live one and copying when needed at point of use.